misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/AIDataMapperCodeActionUtil.java

```diff
@@ -69,42 +68,33 @@
     /**
      * Returns the workspace edits for the automatic data mapping code action.
      *
-     * @param context     {@link LSContext}
-     * @param refAtCursor {@link SymbolReferencesModel.Reference}
-     * @param diagnostic  {@link Diagnostic}
+     * @param context    {@link LSContext}
+     * @param positionDetails {@link PositionDetails}
+     * @param diagnostic {@link Diagnostic}
      * @return edits for the data mapper code action
      * @throws IOException                throws if error occurred when getting generatedRecordMappingFunction
-     * @throws WorkspaceDocumentException throws if error occurred when reading file content
      */
     static List<TextEdit> getAIDataMapperCodeActionEdits(CodeActionContext context,
-                                                         SymbolReferencesModel.Reference refAtCursor,
+                                                         PositionDetails positionDetails,
                                                          Diagnostic diagnostic)
-            throws IOException, WorkspaceDocumentException {
+            throws IOException {
         List<TextEdit> fEdits = new ArrayList<>();
         String diagnosticMessage = diagnostic.getMessage();
         Matcher matcher = CommandConstants.INCOMPATIBLE_TYPE_PATTERN.matcher(diagnosticMessage);
 
         if (!(matcher.find() && matcher.groupCount() > 1)) {
             return fEdits;
         }
-        String foundTypeLeft = matcher.group(1)
-                .replaceAll(refAtCursor.getSymbol().type.tsymbol.pkgID.toString() + ":",
-                        "");  // variable at left side of the equal sign
-        String foundTypeRight = matcher.group(2)
-                .replaceAll(refAtCursor.getSymbol().type.tsymbol.pkgID.toString() + ":",
-                        "");  // variable at right side of the equal sign
+
+        String foundTypeLeft = matcher.group(1);
+        String foundTypeRight = matcher.group(2);
 
         // Insert function call in the code where error is found
-        BLangNode bLangNode = refAtCursor.getbLangNode();
-        Position startPos = new Position(bLangNode.pos.lineRange().startLine().line() - 1,
-                bLangNode.pos.lineRange().startLine().offset() - 1);
-        Position endPosWithSemiColon = new Position(bLangNode.pos.lineRange().endLine().line() - 1,
-                bLangNode.pos.lineRange().endLine().offset());
-        Range newTextRange = new Range(startPos, endPosWithSemiColon);
-
-        BSymbol symbolAtCursor = refAtCursor.getSymbol();
+        Range newTextRange = diagnostic.getRange();
+
+        String symbolAtCursor = context.workspace().semanticModel(context.filePath()).get().symbol(context.filePath().getFileName().toString(), LinePosition.from(context.getCursorPosition().getLine(), context.getCursorPosition().getCharacter())).get().name();
```

Comment:
This seems to be longer than 120 chars. Please format properly

Reply:
Formated the code.